src/app/_components/post-body.tsx

```
@@ -1,16 +1,62 @@
+    const parts = [];
+    let lastIndex = 0;
+    let match;
+
+    while ((match = mermaidRegex.exec(html)) !== null) {
+      // Add content before the mermaid div
+      if (match.index > lastIndex) {
+        parts.push(
+          <div
+            key={`content-${lastIndex}`}
+            className={markdownStyles["markdown"]}
+            dangerouslySetInnerHTML={{ __html: html.slice(lastIndex, match.index) }}
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)